src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx

```diff
@@ -1,11 +1,223 @@
+        />
+        <select
+          value={categoryFilter}
+          onChange={(e) => {
+            setCategoryFilter(e.target.value);
+          }}
+          className="p-2 mb-4 border border-gray-300 rounded-md ml-4"
+        >
+          <option value="">All Categories</option>
+          {categories.map((category, index) => (
+            <option key={index} value={category}>
+              {category}
+            </option>
+          ))}
+        </select>
+        <table className="w-full">
+          <thead>
+            <tr>
+              <th className="px-6 py-3 bg-gray-50 text-left text-xs leading-4 font-medium text-gray-500 tracking-wider">
+                Name
+              </th>
+              <th className="px-6 py-3 bg-gray-50 text-left text-xs leading-4 font-medium text-gray-500 tracking-wider">
+                Profession
+              </th>
+              <th className="px-6 py-3 bg-gray-50 text-left text-xs leading-4 font-medium text-gray-500 tracking-wider">
+                Category
+              </th>
+              <th className="px-6 py-3 bg-gray-50 text-left text-xs leading-4 font-medium text-gray-500 tracking-wider">
+                Status
+              </th>
+            </tr>
+          </thead>
+          <tbody className="bg-white">
+            {filteredMentorsByName.map((mentor) => (
+              <tr key={mentor.uuid}>
+                <td className="px-6 py-4 whitespace-no-wrap border-b border-gray-200">
+                  {mentor.application.firstName} {mentor.application.lastName}
+                </td>
+                <td className="px-6 py-4 whitespace-no-wrap border-b border-gray-200">
+                  {mentor.application.position}
+                </td>
+                <td className="px-6 py-4 whitespace-no-wrap border-b border-gray-200">
+                  {mentor.category.category}
+                </td>
+                <td className="px-6 py-4 whitespace-no-wrap border-b border-gray-200">
+                  <select
+                    value={mentor.state}
+                    onChange={async (e) => {
+                      await handleStatusUpdate(mentor.uuid, e.target.value);
+                    }}
+                    className="py-2 px-4 border border-gray-300 rounded-md"
+                  >
+                    <option value="pending">Pending</option>
+                    <option value="rejected">Rejected</option>
+                    <option value="approved">Approved</option>
+                  </select>
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </table>
+      </>
+    );
+  };
+
+  const fetchMentors = async () => {
+    const url = `${API_URL}/admin/mentors`;
+    try {
+      const response = await axios.get(url, { withCredentials: true });
+      setMentors(response.data.mentors);
+    } catch (error) {
+      console.error('Error fetching mentors:', error);
+    } finally {
+      setLoading(false);
+    }
+  };
```

Comment:
create a custom hook for this useMentors and use useQuery to fetch them with axios. take a look at the mentor application.

Reply:
Implemented a custom hook `useMentors` as requested. please check.